src/Moq/Mock.cs

```diff
@@ -395,36 +393,53 @@ internal void AddInnerMock(MethodInfo method, object returnValue)
 		{
 			Debug.Assert(Mock.TryGetFromReturnValue(returnValue, out _));
 
-			this.InnerMocks.TryAdd(method, returnValue);
+			var arguments = GetArguments(method);
+			var obj = Expression.Parameter(this.MockedType, "mock");
+			var expression = Expression.Lambda(Expression.Call(obj, method, arguments), obj);
+
+			this.Setups.Add(new FixedReturnValueSetup(method, arguments, expression, returnValue));
+
+			Expression[] GetArguments(MethodInfo m)
+			{
+				var itIsAnyMethod = typeof(It).GetMethod(nameof(It.IsAny), BindingFlags.Public | BindingFlags.Static);
+				var parameterTypes = m.GetParameterTypes();
+				var result = new Expression[parameterTypes.Count];
+				for (int i = 0, n = parameterTypes.Count; i < n; ++i)
+				{
+					result[i] = Expression.Call(itIsAnyMethod.MakeGenericMethod(parameterTypes[i]));
```

Comment:
Hmm. I wonder if there are any edge cases &mdash; nullable parameter types, generic type parameter, etc. &mdash; that could make this fail. This might require some further testing.